### PR TITLE
Swagger Changes to Add DnsZonePartner and Collation into Managed Instance 

### PR DESCRIPTION
**Pull request by ziwa-msft**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ x] The title of the PR is clear and informative.
- [x ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x ] If applicable, the PR references the bug/issue that it fixes.
- [x ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ x] My spec meets the review criteria:
  - [x ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceCreateMax.json [hunk 1]
```diff
@@ -44,7 +46,8 @@
           "fullyQualifiedDomainName" : "testinstance.database.windows.net",
           "administratorLogin" : "dummylogin",
           "subnetId" : "/subscriptions/20D7082A-0FC7-4468-82BD-542694D5042B/resourceGroups/testrg/providers/Microsoft.Network/virtualNetworks/vnet1/subnets/subnet1",
-          "state" : "Ready"
+          "state": "Ready",
+          "dnsZone":  "1b4e2caff2530"
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceCreateMax.json]: Is this a realistic-looking dns zone?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceList.json [hunk 2]
```diff
@@ -22,7 +22,8 @@
               "state": "Ready",
               "vCores": 8,
               "storageSizeInGB": 1024,
-              "licenseType": "Full"
+              "licenseType": "Full",
+              "collation":  "SQL_Latin1_General_CP1_CI_AS"
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceList.json]: Looks like dnsZone is missing from all these other examples

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json [hunk 3]
```diff
@@ -351,6 +351,26 @@
           "format": "int32",
           "description": "The maximum storage size in GB.",
           "type": "integer"
+        },
+        "collation": {
+          "description": "Collation of the managed instance.",
+          "type": "string",
+          "x-ms-mutability": [
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json]: I thought this was made readonly now

**Review by jaredmoo — CHANGES_REQUESTED** [review]

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceCreateMax.json (hunk 1 above)
Comment (ziwa-msft) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceCreateMax.json]: Yes it is. We generate dns zone with the first part of a new Guid.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceList.json (hunk 2 above)
Comment (ziwa-msft) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceList.json]: It has been added now

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceList.json [hunk 4]
```diff
@@ -35,12 +37,14 @@
...
-              "fullyQualifiedDomainName" : "testcl2.database.windows.net",
+              "fullyQualifiedDomainName" : "testcl2.1b4e2caff2530.database.windows.net",
               "administratorLogin" : "cloudSA",
               "state": "Ready",
               "vCores": 8,
               "storageSizeInGB": 1024,
-              "licenseType": "Full"
+              "licenseType": "Full",
+              "collation":  "SQL_Latin1_General_CP1_CI_AS",
+			  "dnsZone": "1b4e2caff2530"
```
Comment (jaredmoo) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/examples/ManagedInstanceList.json]: nit: fix indentation

**Review by jaredmoo — APPROVED** [review]
Minor nitpicks, otherwise approved

File: specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json (hunk 3 above)
Comment (storak) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2015-05-01-preview/managedInstances.json]: Yes, the Collation property is made read-only a three weeks ago. Mike, which sync point you've used for generating the swagger changes?

**Review by jaredmoo — APPROVED** [review]

**Review by storak — APPROVED** [review]

**Comment by lmazuel** [comment]
If @jaredmoo is good, I am too ;)

**Comment by jaredmoo** [comment]
Thanks!

**Pull request merged**